### DISCUSSION TITLE
How to load a named template in Lua?

### DISCUSSION THREAD

**Discussion by jeffvalk** [jgm/pandoc#9853] [Q&A] [answered]
How does one load an arbitrarily named template in Lua, assuming the template is in the data directory?

I'm imagining something like the following, which (almost) works:

```lua
  local template_text = pandoc.template.default("my_template.html")
  local options = {template = pandoc.template.compile(template_text)}
  pandoc.write(doc, "html", options)
```

...except that it searches for a template named `default.my_template.html`. I'd like to use one named `my_template.html`.

**Comment by tarleb** [comment, marked as answer]
There's no special function for that, so it's down to basic Lua functions:

``` lua
local path = require 'pandoc.path'
local template_path = path.join{PANDOC_STATE.user_data_dir, 'template', 'my_template.html'}
local template_text = io:open(template_path):read 'a'
pandoc.write(doc, 'html', {template = template_text})
```

**Reply by jeffvalk** [reply]
Thanks @tarleb

A Lua function that resolves resources in the data directory (templates, filters, other user-defined files) might be a simple, ergonomic enhancement. It would feel more "batteries included" in the API, and it would give filter/reader/writer authors an indication that they're working with Pandoc's architecture and not against it.

**Reply by tarleb** [reply]
Agreed, a function named `pandoc.template.file` or `pandoc.template.get` would probably make sense. Would you open an issue for that?

**Reply by jeffvalk** [reply]
Sure. Now in #9854
